### PR TITLE
fix regressions and messages

### PR DESCRIPTION
**Pull request by lipovsek-aws**
Regressed https://github.com/ptoulme-aws/axlearn/pull/25

### CODE REVIEW

File: axlearn/common/flash_attention/neuron_attention_test.py [hunk 1]
```diff
@@ -11,6 +11,10 @@
 from axlearn.common.flash_attention.utils import mha_reference
 
 
+if jax.default_backend() != "neuron":
+    pytestmark = pytest.mark.skip(reason="Incompatible hardware, Neuron only test.")
```
Comment (huntingcarlisle) [inline axlearn/common/flash_attention/neuron_attention_test.py]: We should be more explicit with skip reason message: `"Incompatible hardware, Test only runs on AWS Neuron devices."`

File: axlearn/cloud/gcp/tpu_health_check.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 from contextlib import contextmanager
 from datetime import datetime
 from typing import Literal, Optional, Union
+import pytest
```
Comment (patrick-toulme) [inline axlearn/cloud/gcp/tpu_health_check.py:34]: is this import necessary?

File: axlearn/common/flash_attention/neuron_attention_test.py [hunk 3]
```diff
@@ -11,6 +11,10 @@
 from axlearn.common.flash_attention.utils import mha_reference
 
 
+if jax.default_backend() != "neuron":
+    pytestmark = pytest.mark.skip(reason="Incompatible hardware, AWS Neuron only test.")
+
```
Comment (patrick-toulme) [inline axlearn/common/flash_attention/neuron_attention_test.py:16]: remove blank lines

File: axlearn/common/host_array_test.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 from absl.testing import absltest, parameterized
 from jax import numpy as jnp
 from jax.experimental import mesh_utils
+import pytest
```
Comment (patrick-toulme) [inline axlearn/common/host_array_test.py:15]: is this necessary?

File: axlearn/common/inference_test.py [hunk 5]
```diff
@@ -168,13 +169,10 @@ def predict_batch(self, input_batch: NestedTensor) -> NestedTensor:
 def is_supported(
     platform: str,
     mesh_shape: tuple[int, int],
-    param_dtype: jnp.dtype,
```
Comment (patrick-toulme) [inline axlearn/common/inference_test.py:171]: is this needed?

File: axlearn/common/inference_test.py [hunk 6]
```diff
@@ -168,13 +169,10 @@ def predict_batch(self, input_batch: NestedTensor) -> NestedTensor:
...
     inference_dtype: Optional[jnp.dtype],
     global_batch_size: int,
     data_partition: DataPartitionType,
-    use_ema: bool = False,
 ):
-    del param_dtype, use_ema  # not used
```
Comment (patrick-toulme) [inline axlearn/common/inference_test.py:177]: why are we changing this?

File: axlearn/common/inference_test.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
 from absl.testing import absltest, parameterized
 from jax.experimental import mesh_utils
 from jax.experimental.pjit import pjit
+import pytest
```
Comment (patrick-toulme) [inline axlearn/common/inference_test.py:23]: is this needed?

File: axlearn/cloud/gcp/tpu_health_check.py (hunk 2 above)
Comment (lipovsek-aws) [reply axlearn/cloud/gcp/tpu_health_check.py:34]: https://github.com/ptoulme-aws/axlearn/blob/52d0991ed0224f034ccf82b088d7e39b112cd302/axlearn/cloud/gcp/tpu_health_check.py#L65

Comment (lipovsek-aws) [reply axlearn/cloud/gcp/tpu_health_check.py:34]: we could potentially import thorugh abseil library, would have to check

File: axlearn/common/host_array_test.py (hunk 4 above)
Comment (lipovsek-aws) [reply axlearn/common/host_array_test.py:15]: https://github.com/ptoulme-aws/axlearn/blob/52d0991ed0224f034ccf82b088d7e39b112cd302/axlearn/common/host_array_test.py#L68

File: axlearn/common/inference_test.py (hunk 7 above)
Comment (lipovsek-aws) [reply axlearn/common/inference_test.py:23]: https://github.com/ptoulme-aws/axlearn/blob/52d0991ed0224f034ccf82b088d7e39b112cd302/axlearn/common/inference_test.py#L325

File: axlearn/common/inference_test.py (hunk 5 above)
Comment (lipovsek-aws) [reply axlearn/common/inference_test.py:171]: no, see the function bellow

File: axlearn/common/inference_test.py (hunk 6 above)
Comment (lipovsek-aws) [reply axlearn/common/inference_test.py:177]: we don't need these args to check for skipping

File: axlearn/common/flash_attention/neuron_attention_test.py (hunk 3 above)
Comment (lipovsek-aws) [reply axlearn/common/flash_attention/neuron_attention_test.py:16]: isn't this PEP8? I'll check with linting

**Comment by patrick-toulme** [comment]
Thanks

**Pull request merged**